crypto/mem_sec.c

```diff
@@ -433,8 +437,13 @@ static int sh_init(size_t size, int minsize)
     if (mprotect(sh.map_result + aligned, pgsize, PROT_NONE) < 0)
         ret = 2;
 
+#if defined(MLOCK_ONFAULT) && defined(SYS_mlock2)
+    if (syscall(SYS_mlock2, sh.arena, sh.arena_size, MLOCK_ONFAULT) < 0)
+        ret = 2;
+#else
```

Comment:
What if you compiled with headers X but executed on kernel Y? What I'm hinting at is that it would probably be better to use mlock as fall-back in case SYS_mlock2 fails with [at least?] ENOSYS.